src/main/main.cpp

```diff
@@ -71,15 +71,46 @@ unique_ptr<GPU> create_gpu(MemoryInterface *memory_ptr, CPUInterface *cpu_ptr,
 int main(int argc, char *argv[]) {
 	ios_base::sync_with_stdio(false);
 
+	auto rom_path = string("");
 	if (argc < 2) {
-		cerr << "Please provide the name of a ROM file." << endl;
-		exit(1);
+		cerr << "Please provide the name of a ROM file or press q/Q to exit"
+		     << endl;
+		getline(cin, rom_path);
+		if (rom_path == "q" || rom_path == "Q") {
+			cout << "Exiting TVP" << endl;
+			exit(0);
+		}
+	} else {
+		rom_path = string(argv[1]);
 	}
-	auto rom_path = string(argv[1]);
 
 	auto cartridge = std::make_unique<Cartridge>(rom_path);
+
+	// Reading Meta Data of ROM file and checking if ROM is valid or Not
+	unsigned short int logo_hex_dump[48] = {
```

Comment:
You'll want to move all this logic into the cartridge module, and inside some helper class like I've described above. Also, it would be better to put the raw logo data into some module level header. You can define a `cartridge/include/cartridge/utils.h`, like how it's been done in the `memory` module.

Reply:
Use `uint8_t`